mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +242,81 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionWithInvalidKey() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
+        expected.expect(InvalidAnnotationException.class);
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(INVALID_KEY_ANNOTATION));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionWithNullKey() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
+        expected.expect(NullPointerException.class);
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(MailboxAnnotation.newInstance(null, "value")));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldUpdateStoredAnnotation() throws BadCredentialsException, MailboxException, InvalidAnnotationException {
```

Comment:
MailboxManagerTests are played with maven when called by : 

```
 mvn -Dtest=JCRMailboxManagerTest -DfailIfNoTests=false -pl org.apache.james:apache-james-mailbox-jcr -am test
```

They can be played directly from the IDE (intelliJ as well).

However it do not work when played with : 

```
mvn clean install
```

And display warning like : 

```
[main] INFO org.xenei.junit.contract.ContractSuite - No tests for class org.apache.james.mailbox.jpa.JPAMailboxManagerTest
```

The cool point is that your PR is green even if you did not checked the capability before your test. However, it means I have to investigate on this...